app/lib/features/space/sheets/link_chat_sheet.dart

```diff
@@ -0,0 +1,224 @@
+}
+
+class _LinkChatPageConsumerState extends ConsumerState<LinkChatPage> {
+  final TextEditingController searchTextEditingController =
+      TextEditingController();
+  final List<String> subChatsIds = [];
+
+  @override
+  void initState() {
+    WidgetsBinding.instance.addPostFrameCallback((Duration duration) {
+      selectParentSpaceData();
+    });
+    super.initState();
+  }
+
+  //Select parent space data
+  void selectParentSpaceData() {
+    if (widget.parentSpaceId != null) {
+      final notifier = ref.read(selectedSpaceIdProvider.notifier);
+      notifier.state = widget.parentSpaceId;
+    }
+  }
+
+  //Fetch known sub-chats list of selected parent space
+  void fetchKnownSubChatsData() {
+    final selectedParentSpaceId = ref.watch(selectedSpaceIdProvider);
+    if (selectedParentSpaceId == null) return;
+    final sp = ref.watch(spaceRelationsOverviewProvider(selectedParentSpaceId));
+    sp.when(
+      data: (space) {
+        subChatsIds.clear();
+        for (int i = 0; i < space.knownChats.length; i++) {
+          subChatsIds.add(space.knownChats[i].getRoomId().toString());
+        }
+      },
+      error: (e, s) => Container(),
+      loading: () => Container(),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final size = MediaQuery.of(context).size;
+
+    //Fetch known sub-chats list of selected parent space
+    fetchKnownSubChatsData();
+
+    return SideSheet(
+      header: 'Link Sub-Chat',
+      body: SizedBox(
+        height: size.height - 120,
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.start,
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            parentSpaceSelector(context, ref),
+            Search(
+              onChanged: (value) => ref
+                  .read(chatSearchValueProvider.notifier)
+                  .update((state) => value),
+              searchController: searchTextEditingController,
+            ),
+            Expanded(child: chatsList(context, ref)),
+          ],
+        ),
+      ),
+    );
+  }
+
+  //Parent space selector
+  Widget parentSpaceSelector(BuildContext context, WidgetRef ref) {
+    return const Padding(
+      padding: EdgeInsets.all(16),
+      child: SelectSpaceFormField(
+        canCheck: 'CanLinkSpaces',
+        mandatory: true,
+        title: 'Parent space',
+        emptyText: 'optional parent space',
+        selectTitle: 'Select parent space',
+      ),
+    );
+  }
+
+  //List of chats excluding DMs that can be linked according to the selected parent space
+  Widget chatsList(BuildContext context, WidgetRef ref) {
+    final searchValue = ref.watch(chatSearchValueProvider);
+    if (searchValue != null && searchValue.isNotEmpty) {
+      return ref.watch(searchedChatsProvider).when(
+            data: (chats) {
+              var chatList =
+                  chats.where((element) => (!element.isDm())).toList();
```

Comment:
I feel like this matching should already be done in the provider ...

Reply:
Sorry but I was not able to find matching filter on the Chat Providers. Can you help me with the name of similar matching provider?

Reply:
just create a new provider...